lexer/lexer.py

```diff
@@ -68,7 +69,7 @@ def tokenize(self, line):
         # The token values will be the actual values of the tokens
 
         # split the line into tokens by spaces eg "1 + 4" -> ["1", "+", "4"]
-        self.tokens = line.split()
+        self.tokens = re.findall(r"\d+\.\d+|\d+|[a-zA-Z0-9]+|[+\-*/%()=]", line)
```

Comment:
<img width="598" alt="image" src="https://github.com/sharmaigne/SNOL/assets/71376956/8cd33265-5e2b-4a3a-a3ab-6f7293b67ed8">

Lexer should recognize "4." as a float, not an integer.

This is how python is by default and it's also specified in the program details with the EBNF rule:
`float := digit{digit}.{digit}`